jdaviz/tests/test_app.py

```diff
@@ -1,142 +1,143 @@
-    assert dc[3].label == "test (2)"
-
-
-def test_duplicate_data_labels_with_brackets(specviz_helper, spectrum1d):
-    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
-    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
-    dc = specviz_helper.app.data_collection
-    assert len(dc) == 2
-    assert dc[0].label == "test[test]"
-    assert dc[1].label == "test[test] (1)"
-
-
-def test_return_data_label_is_none(specviz_helper):
-    data_label = specviz_helper.app.return_data_label(None)
-    assert data_label == "Unknown"
-
-
-def test_return_data_label_is_image(specviz_helper):
-    data_label = specviz_helper.app.return_data_label("data/path/test.jpg")
-    assert data_label == "test[jpg]"
-
-
-def test_hdulist_with_filename(cubeviz_helper, image_cube_hdu_obj):
-    image_cube_hdu_obj.file_name = "test"
-    data_label = cubeviz_helper.app.return_data_label(image_cube_hdu_obj)
-    assert data_label == "test[HDU object]"
-
-
-def test_file_path_not_image(imviz_helper, tmp_path):
-    path = tmp_path / "myimage.fits"
-    path.touch()
-    data_label = imviz_helper.app.return_data_label(str(path))
-    assert data_label == "myimage"
-
-
-def test_unique_name_variations(specviz_helper, spectrum1d):
-    data_label = specviz_helper.app.return_unique_name(None)
-    assert data_label == "Unknown"
-
-    specviz_helper.load_spectrum(spectrum1d, data_label="test[flux]")
-    data_label = specviz_helper.app.return_data_label("test[flux]", ext="flux")
-    assert data_label == "test[flux][flux]"
-
-    data_label = specviz_helper.app.return_data_label("test", ext="flux")
-    assert data_label == "test[flux] (1)"
-
-
-def test_substring_in_label(specviz_helper, spectrum1d):
-    specviz_helper.load_spectrum(spectrum1d, data_label="M31")
-    specviz_helper.load_spectrum(spectrum1d, data_label="M32")
-    data_label = specviz_helper.app.return_data_label("M")
-    assert data_label == "M"
-
-
-@pytest.mark.parametrize('data_label', ('111111', 'aaaaa', '///(#$@)',
-                                        'two  spaces  repeating',
-                                        'word42word42word  two  spaces'))
-def test_edge_cases(specviz_helper, spectrum1d, data_label):
-    dc = specviz_helper.app.data_collection
-
-    specviz_helper.load_spectrum(spectrum1d, data_label=data_label)
-    specviz_helper.load_spectrum(spectrum1d, data_label=data_label)
-    assert dc[1].label == f"{data_label} (1)"
-
-    specviz_helper.load_spectrum(spectrum1d, data_label=data_label)
-    assert dc[2].label == f"{data_label} (2)"
-
-
-def test_case_that_used_to_break_return_label(specviz_helper, spectrum1d):
-    specviz_helper.load_spectrum(spectrum1d, data_label="this used to break (1)")
-    specviz_helper.load_spectrum(spectrum1d, data_label="this used to break")
-    dc = specviz_helper.app.data_collection
-    assert dc[0].label == "this used to break (1)"
-    assert dc[1].label == "this used to break (2)"
+import pytest
```

Comment:
I'm having a hard time figuring out what changed in this file, can you point me to the actual diff?

Reply:
Ugh... this happened again. I don't know why GitHub does this sometimes. Let me see if I can track it down (git diff on my machine is showing the same thing here)

Reply:
@javerbukh This is the only thing that changed in that file: https://github.com/spacetelescope/jdaviz/pull/2242/commits/92c0bd8f7d99a8cf7e2de38ec3f2337d09b0468b

You can see the diff here properly! I don't know why GitHub is showing the entire file as changed...

Reply:
It looks like your editor is either changing the newline or stripping out trailing whitespace on save.  Can you stage and commit only the individual lines instead of all changes?  (I'm guessing the individual commit diff is set to ignore whitespace changes).

Reply:
I don't think this is the case. Whenever I commit, there's an explicit line that converts my MSDOS line endings to UNIX line endings:
`dos2unix: converting file E:/STScI/gitRepos/jdaviz/.git/COMMIT_EDITMSG to Unix format...`

Because WYSI NOT WYG on my end, I can't really inspect what's getting committed. Ironically, I think the way to check for that is someone on the mac side open the file and see if you have a CR or LF character at the end of the line (But that being said, if it is a line character thing, I wouldn't expect the OSX or Ubuntu tests to successfully run at all

Reply:
Probably the endlines. I think we have mixed endlines in the code base. I ended up blinking the copy from this branch with the copy from `main` and this is the only difference:

```python
    assert not len(viz.app.get_data_from_viewer("h", "non-existent label"))
```

```python
    with pytest.raises(ValueError):
        viz.get_data("non-existent label")
```

I think the behavior change is acceptable. The new behavior is more explicit.

Reply:
No I mean this PR touching all of `test_app.py`. I'd prefer to maintain the correct git history if possible.

Reply:
Errr... I can try but I cannot promise anything.